gsy_framework/schema/validators.py

```diff
@@ -181,3 +197,12 @@ def deserialize(self, data: Dict) -> Dict:
         data["aggregator_device_mapping"] = self._aggregator_mapping_serializer.deserialize(
             data["aggregator_device_mapping"])
         return data
+
+
+def get_schema_validator(schema_name: str) -> BaseSchemaValidator:
+    """Return the appropriate schema validator class based on the schema name."""
+    if schema_name == "launch_simulation_settings":
+        return AVROSimulationSettingsSerializer()
+    if schema_name == "launch_simulation":
+        return SimulationLaunchSerializer()
+    return AVROSchemaSerializer(schema_name=schema_name)
```

Comment:
I understand that this was requested in the PR comments, but I do not understand the benefits of this abstraction. This method will gain if blocks when we add serialisers. I personally preferred the version before where specific Serializers where selected in the calling class. Especially the different interfaces of the Serializers make this method not worth it to me. 
Only a minor doubt, no need to convince me, but maybe you have a good reason for this that you could share and help me understand. Thanks a lot!

Reply:
I take the opportunity to answer this question.
If we imagine the `Ports and Adapters a.k.a. Hexagonal` architecuture as a city, the factory function would be the door and key point of this city. With the help of this function, we can change the whole serialization/validation system in a matter of minutes (or choose the correct serializer dynamically by simply updating this function). As a rule of thump, we never want to see occurances of our custom classes in our code and the code should only know about the interface. A good example of this would be the `django.contrib.auth.get_user_model` which returns the correct user model by checking our settings module.

Reply:
I take the opportunity to answer this question.
If we imagine the Ports and Adapters a.k.a. Hexagonal architecuture as a city, the factory function would be the door and key point of this city. With the help of this function, we can change the whole serialization/validation system in a matter of minutes (or choose the correct serializer dynamically by simply updating this function). As a rule of thump, we never want to see occurances of our custom classes in our code and the code should only know about the interface. A good example of this would be the django.contrib.auth.get_user_model which returns the correct user model by checking our settings module.

Reply:
I take the opportunity to answer this question.
If we imagine the Ports and Adapters a.k.a. Hexagonal architecuture as a city, the factory function would be the door and key point of this city. With the help of this function, we can change the whole serialization/validation system in a matter of minutes (or choose the correct serializer dynamically by simply updating this function). As a rule of thump, we never want to see occurances of our custom classes in our code and the code should only know about the interface. A good example of this would be the django.contrib.auth.get_user_model which returns the correct user model by checking our settings module.

Reply:
Just to put my feedback here too, factory classes have the advantage that you have one entry point for all parts of the code. This means that the caller code does not need to select the class that it needs to instantiate, but only to know how to use this method. That way it is far easier to create / use the serializer no matter the implementation or the protocol used as Amirhossein describes. Also it makes it much more easy to test, because only a single method needs to be mocked instead of many different classes. 
Personally what I do not like in this implementation is that the dispatching is done via strings and not by e.g. an enum, because this requires knowledge about the exact string that needs to be sent to the `get_schema_validator` method. Nevertheless I thought to skip it in order to not overcomplicate the PR. If you would like this approach though let me know to adapt it, thanks!